### ISSUE TITLE
Create rule XXX: Switch statements should be focused

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#4421]
#### Detection

- Forbid loops and declarations like functions and classes
- Allow anonymous functions
- Beware of the overlap with [S6836](https://sonarsource.github.io/rspec/#/rspec/S6836)

#### Noncompliant

```js
switch (x) {
    case 42:
        function foo() { /* ... */ } // Noncompliant
        foo();
        break; 
    case 43:
        for (let i = 0; i < 5; i++) { // Noncompliant
            /* ... */
        }
        break;
}
```

#### Compliant

```js
function foo() { /* ... */ }
function bar() {
    for (let i = 0; i < 5; i++) {
        /* ... */
    }
}

switch (x) {
    case 42:
        foo();
        break; 
    case 43:
        bar();
        break;
}
```

**Comment by Wohops** [comment]
Migrated to Jira. New ticket number: JS-61
